### PR TITLE
Raven default to pessimistic concurrency

### PR DESCRIPTION
**Pull request by lailabougria**
Documentation for https://github.com/Particular/NServiceBus.RavenDB/pull/611

### CODE REVIEW

File: persistence/ravendb/saga-concurrency_pessimistic-locking_raven_[7,).partial.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+## Sagas concurrency control
+
+The RavenDB persistence allows for both optimistic and pessimistic concurrency options. The default, starting from v7, is pessimistic concurrency.
+
+TODO: do we want to add any guidance on when to opt for optimistic locking?
```
Comment (lailabougria) [inline persistence/ravendb/saga-concurrency_pessimistic-locking_raven_[7,).partial.md:5]:
```suggestion
```

**Comment by lailabougria** [comment]
@mauroservienti @kbaley This one is still awaiting your seal of approval!

**Comment by kbaley** [comment]
I think we need to update [the "At a glance"](https://docs.particular.net/persistence/ravendb/) page as well

File: persistence/ravendb/index_glance_raven_[6.5,).partial.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+For a description of each feature, see the [persistence at a glance legend](/persistence/#persistence-at-a-glance).
+
+|Feature                    |   |
+|:---                       |---
+|Supported storage types    |Sagas, Outbox, Subscriptions, Timeouts
+|Transactions               |via `IDocumentSession.SaveChanges()`
+|Concurrency control        |Pessimistic concurrency, optional optimistic concurrency
```
Comment (kbaley) [inline persistence/ravendb/index_glance_raven_[6.5,).partial.md:7]: I thought in 6.5 optimistic concurrency was still the default but we just added a new API and marked the old one obsolete.

Comment (mauroservienti) [reply persistence/ravendb/index_glance_raven_[6.5,).partial.md:7]: What @kbaley said. We probably need the partial file to be `index_glance_raven_[6.5,7.0).partial.md` and then another partial `index_glance_raven_[7.0,).partial.md` that has "Pessimistic concurrency, optional optimistic concurrency"

File: persistence/ravendb/index_glance_raven_[,6.5).partial.md [hunk 3]
```diff
@@ -0,0 +1,9 @@
+For a description of each feature, see the [persistence at a glance legend](/persistence/#persistence-at-a-glance).
```
Comment (mauroservienti) [inline persistence/ravendb/index_glance_raven_[,6.5).partial.md:1]: This file should be named `index_glance_raven_[,7.0).partial.md`

File: persistence/ravendb/saga-concurrency_concurrency_raven_[6.4,7).partial.md [hunk 4]
```diff
@@ -1 +1,2 @@
-By default, RavenDB persistence uses [optimistic concurrency control](https://en.wikipedia.org/wiki/Optimistic_concurrency_control) when updating or deleting saga data, though starting with NServiceBus.RavenDB version 6.4, it's possible to consider the persister to use pessmimistic locking. See [later in this document](#sagas-pessimistic-locking) for how to do this.
\ No newline at end of file
+By default, RavenDB persistence uses [optimistic concurrency control](https://en.wikipedia.org/wiki/Optimistic_concurrency_control) when updating or deleting saga data, though starting with NServiceBus.RavenDB version 6.4, it's possible to consider the persister to use pessmimistic locking.
```
Comment (mauroservienti) [inline persistence/ravendb/saga-concurrency_concurrency_raven_[6.4,7).partial.md:1]:
```suggestion
By default, RavenDB persistence uses [optimistic concurrency control](https://en.wikipedia.org/wiki/Optimistic_concurrency_control) when updating or deleting saga data, though starting with NServiceBus.RavenDB version 6.4, it's possible to configure the persister to use pessimistic locking.
```

File: persistence/ravendb/saga-concurrency_pessimistic-locking_raven_[6.4,6.5).partial.md [hunk 5]
```diff
@@ -10,7 +10,7 @@ Choose pessimistic locking over optimistic locking if the system is experiencing
```
Comment (mauroservienti) [inline persistence/ravendb/saga-concurrency_pessimistic-locking_raven_[6.4,6.5).partial.md:10]: The range in the file name should be `[6.5,7.0)`

**Review by mauroservienti — APPROVED** [review]

**Review by kbaley — APPROVED** [review]

**Pull request merged**